services/js/src/main/scala/org/apache/gearpump/dashboard/DashboardApp.scala

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gearpump.dashboard
+
+import com.greencatsoft.angularjs.Angular
+import org.apache.gearpump.dashboard.services._
+
+import scala.scalajs.js.JSApp
+import scala.scalajs.js.annotation.JSExport
+
+@JSExport
+object DashboardApp extends JSApp {
```

Comment:
Angular HttpService wraps the JavaScript $http code which is what restapi.js calls. Then shouldn't restapi.js also use XmlHttpRequest? Why are we forcing non angular use for RestApiService.scala but are ok with $http for restapi.js? This may introduce many subtle bugs.



> On Aug 3, 2015, at 12:21 AM, Sean Zhong <notifications@github.com> wrote:
> 
> In services/js/src/main/scala/org/apache/gearpump/dashboard/DashboardApp.scala:
> 
> > + * distributed under the License is distributed on an "AS IS" BASIS,
> > + * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
> > + * See the License for the specific language governing permissions and
> > + * limitations under the License.
> > + */
> > +
> > +package org.apache.gearpump.dashboard
> > +
> > +import com.greencatsoft.angularjs.Angular
> > +import org.apache.gearpump.dashboard.services._
> > +
> > +import scala.scalajs.js.JSApp
> > +import scala.scalajs.js.annotation.JSExport
> > +
> > +@JSExport
> > +object DashboardApp extends JSApp {
> @kkasravi
> 
> The RestService now depends on greencatsoft HttpService. I think for this point, we should avoid introducing any greencatsoft Angular stuff, until the discussion is clear.
> 
> For RestApiService, you can choose to use XmlHttpRequest to do ajax call. 
> Or 
> http://lihaoyi.github.io/hands-on-scala-js/#dom.extensions
> 
> —
> Reply to this email directly or view it on GitHub.
>